pure2.md

```diff
@@ -0,0 +1,535 @@
+* Directing/specifying a specific bundle of images to be installed is a very important safety, security, and regulatory concern in multi-ECU systems; using a TUF-only approach relies on the person installing the updates to make correct compatibility choices.
+
+### Understanding of device fleets' possible install base
+
+Any offline update scheme must deal with the fact that the Uptane server will not have perfectly up-to-date information on the firmware versions of devices in the field. Currently, Uptane implementors are left without guidance on how offline updates should be implemented, and the result of this gap is a patchwork of security measures (or non-measures). It is still relatively common for vehicle ECUs to have a flashing mode secured by nothing more than the inconvenience of physically accessing a particular jumper, and signature checking when flashing firmware via diagnostic tool is still quite uncommon. The practical upshot of this is that operators of Uptane systems must consider the possibility that any arbitrary software could be installed out-of-band.
+
+PURE-2 limits the scope of what may have been installed on a device. If the Uptane client on the device supports PURE-2, it is much more feasible to completely lock down alternate methods of flashing: removing JTAG headers, removing alternate low-level flashing routines from firmware, etc. Because the Offline Update role signs all metadata authorizing updates outside of the normal online process, the scope of possibility of what may be installed in the field can be greatly reduced. Furthermore, because the offline updates are being facilitated by the same client and within the same framework as the OTA updates, it becomes much easier to maintain an accurate and complete history of software updates on the device.
+
+### Revocation of metadata
+
+Like all Uptane metadata, offline-update metadata can be revoked both implicitly (by allowing it to expire) and explicitly (by issuing new metadata superseding and invalidating the old). Explicit revocations will reach devices when they check for an online update, because they always download the latest version of the offline-update snapshot metadata during an update check. However, if the device does not check for updates for a long time, it will not be aware of newer metadata that includes the revocation.
+
+### Validation and updating of metadata
+
+An offline-update image also includes metadata for other Uptane roles:
```

Comment:
The image itself contains metadata? Like, hardcoded updates to the offline-update metadata, or what? I'm confused.

Reply:
Yeah, we have some terminology stuff to take care of here. "Image" here refers to the complete bundle of stuff you stick in the well-known location for the update client to find, not to some particular binary you're installing. That's probably not ideal. Maybe "bundle"?

Reply:
"Bundle" sounds pretty good, as long as that term is defined somewhere.

Reply:
I looked through the document, and there was only one other place where I'd used "image" like this, so I elected to replace it with the word "bundle", and define it parenthetically in-line in both places.